lib/matplotlib/figure.py

```diff
@@ -122,7 +124,7 @@ def add(self, key, a):
         try:
             hash(key)
         except TypeError:
-            raise ValueError("first argument, %s, is not a valid key" % key)
```

Comment:
But transforms are explicitly mutable (not just by fiddling with the attributes: they actually expose a bunch of methods explicitly for that purpose), which means that trying to key on them is bound to give wrong results (this is the same reason why lists are not hashable in python).  (In fact I had prepared another PR for making all transform classes hashable, but that is not a robust solution in the presence of mutability.)

I argued in #7377 that this behavior (trying to detect and drop duplicates axes) is already brittle anyways and should be deprecated.